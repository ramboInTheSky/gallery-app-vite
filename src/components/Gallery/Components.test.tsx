/** @jsxImportSource @emotion/react */
import { render } from "@testing-library/react";
import {
  Container,
  GalleryGrid,
  GalleryCard,
  GalleryImage,
  Author,
  SkeletonCard,
  Sentinel,
} from "./Components";

describe("Styled Components Snapshots", () => {
  it("renders Container correctly", () => {
    const { asFragment } = render(<Container>Content</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders GalleryGrid correctly", () => {
    const { asFragment } = render(<GalleryGrid />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders GalleryCard correctly", () => {
    const { asFragment } = render(<GalleryCard>Card Content</GalleryCard>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders GalleryImage correctly", () => {
    const { asFragment } = render(
      <GalleryImage src="example.jpg" alt="Example" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Author correctly", () => {
    const { asFragment } = render(<Author>Author Name</Author>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders SkeletonCard correctly", () => {
    const { asFragment } = render(<SkeletonCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Sentinel correctly", () => {
    const { asFragment } = render(<Sentinel />);
    expect(asFragment()).toMatchSnapshot();
  });
});
