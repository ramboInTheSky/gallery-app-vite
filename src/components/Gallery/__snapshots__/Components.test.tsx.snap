// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Components Snapshots renders Author correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 0.5rem 1rem;
  font-size: 1rem;
  font-weight: 500;
  text-align: center;
  color: var(--purple);
  background-color: #f9f9f9;
}

<p
    class="emotion-0"
  >
    Author Name
  </p>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders Container correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
  width: 100%;
  padding-top: 1rem;
}

<div
    class="emotion-0"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders GalleryCard correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #fff;
  border: 1px solid #ddd;
  border-radius: 8px;
  overflow: hidden;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  -webkit-transition: box-shadow 0.3s ease;
  transition: box-shadow 0.3s ease;
  min-height: 215px;
}

.emotion-0:hover {
  box-shadow: 0 4px 8px var(--blue);
}

<div
    class="emotion-0"
  >
    Card Content
  </div>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders GalleryGrid correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 1rem;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders GalleryImage correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  height: auto;
  object-fit: cover;
}

<img
    alt="Example"
    class="emotion-0"
    src="example.jpg"
  />
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders Sentinel correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  text-align: center;
  font-size: 0.9rem;
  padding: 1rem;
  color: #555;
}

.emotion-0::after {
  content: "âŒ›";
  display: inline-block;
  margin-left: 0.5rem;
  -webkit-animation: spin 1s linear infinite;
  animation: spin 1s linear infinite;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders SkeletonCard correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 220px;
  background: linear-gradient(90deg, #f0f0f0 25%, #e2e2e2 50%, #f0f0f0 75%);
  -webkit-animation: shimmer 1.5s infinite;
  animation: shimmer 1.5s infinite;
  aria-label: "Loading content";
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
